core/network/src/test/mod.rs

```diff
@@ -342,6 +344,34 @@ impl<D, S: NetworkSpecialization<Block>> Peer<D, S> {
 	pub fn network_service(&self) -> &Arc<NetworkService<Block, S, <Block as BlockT>::Hash>> {
 		&self.network.service()
 	}
+
+	/// Test helper to compare the blockchain state of multiple (networked)
+	/// clients.
+	/// Potentially costly, as it creates in-memory copies of both blockchains in order
+	/// to compare them. If you have easier/softer checks that are sufficient, e.g. 
+	/// by using .info(), you should probably use it instead of this.
+	pub fn blockchain_canon_equals(&self, other: &Self) -> bool {
+		if let (Some(mine), Some(others)) = (self.backend.clone(), other.backend.clone()) {
+			mine.as_in_memory().blockchain()
+				.canon_equals_to(others.as_in_memory().blockchain())
+		} else {
+			false
+		}
+	}
+
+	/// Count the current number of known blocks. Note that:
+	///  1. this might be expensive as it creates an in-memory-copy of the chain
+	///     to count the blocks, thus if you have a different way of testing this
+	///     (e.g. `info.best_hash`) - use that.
+	///  2. this is not reliably always increasing nor effectily acurate as the
+	///     orphaned and proofen-to-never-finalized blocks may be pruned at any time
+	///     thus this number can also drop again.
+	pub fn blocks_count(&self) -> usize {
+		match self.backend {
```

Comment:
`self.backend.as_ref().map(|b| b.as_in_memory().blockchain().blocks_count()).unwrap_or(0)`

Reply:
isn't the match a bit more readable though? Or is this just because of the line breaks?